providers/azure/resources/azure.lr

```diff
@@ -1538,6 +1538,8 @@ private azure.subscription.keyVaultService.key @defaults("kid keyName") {
   version() string
   // List of key versions
   versions() []azure.subscription.keyVaultService.key
+  // Auto-rotation enabled status
```

Comment:
could we make this an optional field and only call the api if needed? (add the parens here and move the call to a function)

Reply:
If you do keep it like it is, please change the description to:

```suggestion
  // Whether auto-rotation is enabled
```

Thanks!

Reply:
@vjeffrey I did as you requested, however I still could not solve the uniqueness of the keys:

![Screenshot from 2024-09-04 14-30-09](https://github.com/user-attachments/assets/ddc8efcd-6f0c-4167-9092-5d0e2a708510)

Reply:
oh, is there maybe an id function missing? i haven't re-checked the code, i'll look when i get online in a while, but usually that's an id thing

Reply:
Thanks VJ, I already added id function as here:
```
func (a *mqlAzureSubscriptionKeyVaultServiceKeyAutorotation) keyName() (string, error) {
	id := a.Kid.Data
	kvid, err := parseKeyVaultId(id)
	if err != nil {
		return "", err
	}

	return kvid.Name, nil
}
```

Reply:
the function should be named `id()` for this to work. alternatively you can just pass in the id directly in the args as `__id`

Reply:
@preslavgerchev  awesome, Thanks. It works now. Appreciate you help.